src/base/bittorrent/torrentimpl.cpp

```diff
@@ -2046,10 +2046,16 @@ void TorrentImpl::manageIncompleteFiles()
         return;
     }
 
+    const libtorrent::file_storage &nativeFiles = m_nativeHandle.torrent_file()->files();
```

Comment:
accessing torrent_file seems to require a lock, so I cache it outside the loop and don't use TorrentImpl::actualPath. I don't know if it's safe but seems to work fine.

Reply:
>accessing torrent_file seems to require a lock, so I cache it outside the loop and don't use TorrentImpl::actualPath.

👍 
>Don't rename file because of difference in path separator

It should go away as soon as #15915 is merged. Otherwise, we would need to fix `TorrentImpl::actualFilePath()` so that it returns uniform paths.

Reply:
keep it for a more direct fix? it can also be used in backporting.

Reply:
> Otherwise, we would need to fix TorrentImpl::actualFilePath() so that it returns uniform paths.

it will not be used here anymore, so that's a story for another day.

Reply:
>keep it for a more direct fix?

It fixes repeated access to `torrent_file()` in loop so it makes sense anyway.
>it can also be used in backporting.

v4.4.x doesn't have the same logic to fix it, doesn't it?

Reply:
> v4.4.x doesn't have the same logic to fix it, doesn't it?

I don't know much about v4.4.x branch.

Reply:
> it will not be used here anymore, so that's a story for another day.

But it is used in `TorrentImpl::handleFileCompletedAlert()` so it can still performs unnecessary renaming due to different separators. In any case, it will be fixed by #15915.